Sources/StreamChat/Controllers/ChannelController/ChannelController.swift

```diff
@@ -1458,16 +1458,31 @@ private extension ChatChannelController {
     }
 
     private func getFirstUnreadMessageId() -> MessageId? {
-        guard let lastReadMessageId = channel?.reads.first(where: { $0.user.id == client.currentUserId })?.lastReadMessageId else {
-            // We default to the oldest message in the history
-            return messages.last?.id
+        let unreads = channel?.reads.filter { $0.unreadMessagesCount > 0 } ?? []
+
+        // If there are no unreads, then return nil.
+        if unreads.isEmpty {
+            return nil
+        }
+
+        let currentUserUnread = unreads.first(where: {
+            $0.user.id == client.currentUserId
+        })
+
```

Comment:
The idea would be that you first get the user's "read", and then you apply the rest of the logic.

```
guard let currentUserRead = channel?.reads.first(where: { $0.user.id == client.currentUserId }) else { 
    return .last(where: { $0.type == .regular || $0.type == .reply })?.id
}

guard currentUserUnread.unreadMessagesCount > 0 else { return nil }

guard let lastReadMessageId = currentUserUnread.lastReadMessageId else { 
    return .last(where: { $0.type == .regular || $0.type == .reply })?.id
}
```

Reply:
Makes sense 👍